mm2src/lp_ordermatch/simple_market_maker.rs

```diff
@@ -0,0 +1,736 @@
+
+    let req = SetPriceReq {
+        base: cfg.base.clone(),
+        rel: cfg.rel.clone(),
+        price: calculated_price,
+        max: cfg.max.unwrap_or(false),
+        volume,
+        min_volume: min_vol,
+        cancel_previous: true,
+        base_confs: cfg.base_confs,
+        base_nota: cfg.base_nota,
+        rel_confs: cfg.rel_confs,
+        rel_nota: cfg.rel_nota,
+        save_in_history: true,
+    };
+
+    let resp = create_maker_order(&ctx, req)
+        .await
+        .map_to_mm(OrderProcessingError::OrderUpdateError)?;
+    info!("Successfully placed order for {} - uuid: {}", key_trade_pair, resp.uuid);
+    Ok(true)
+}
+
+async fn process_bot_logic(ctx: &MmArc) {
+    let simple_market_maker_bot_ctx = TradingBotContext::from_ctx(ctx).unwrap();
+    // note: Copy the cfg here will not be expensive, and this will be thread safe.
+    let cfg_guard = simple_market_maker_bot_ctx.trading_bot_cfg.lock().await;
+    let cfg = cfg_guard.clone();
+    drop(cfg_guard);
+    let rates_registry = match fetch_price_tickers().await {
+        Ok(model) => {
+            info!("price successfully fetched");
+            model
+        },
+        Err(err) => {
+            error!("error during fetching price: {:?}", err);
+            cancel_pending_orders(ctx, &cfg).await;
+            return;
+        },
+    };
+
+    let mut memoization_pair_registry: HashSet<String> = HashSet::new();
+    let ordermatch_ctx = OrdermatchContext::from_ctx(ctx).unwrap();
+    let maker_orders_guard = ordermatch_ctx.my_maker_orders.lock().await;
+    let maker_orders = maker_orders_guard.clone();
+    drop(maker_orders_guard);
+
+    info!("nb_orders: {}", maker_orders.len());
+
+    let mut futures_order_update = Vec::with_capacity(0);
+    // Iterating over maker orders and update order that are present in cfg as the key_trade_pair e.g KMD/LTC
+    for (uuid, value) in maker_orders.into_iter() {
+        let key_trade_pair = TradingPair::new(value.base.clone(), value.rel.clone());
+        match cfg.get(&key_trade_pair.as_combination()) {
+            Some(coin_cfg) => {
+                let (result_sender, result_receiver) = futures::channel::oneshot::channel();
+                let cloned_infos = (
+                    ctx.clone(),
+                    rates_registry
+                        .get_cex_rates(coin_cfg.base.clone(), coin_cfg.rel.clone())
+                        .unwrap_or_default(),
+                    key_trade_pair.clone(),
+                    coin_cfg.clone(),
+                );
+                spawn(execute_update_order(uuid, value, result_sender, cloned_infos));
+                futures_order_update.push(result_receiver);
+                memoization_pair_registry.insert(key_trade_pair.as_combination());
+            },
+            _ => continue,
+        }
+    }
+
+    let all_updated_orders_tasks = futures::future::join_all(futures_order_update.into_iter());
```

Comment:
This is the same as joining `execute_update_order` futures directly. Is there a specific reason for using the additional channel here?

Reply:
> This is the same as joining execute_update_order futures directly. Is there a specific reason for using the additional channel here?

I'm not sure I understood the question, but I'm looking to join futures once all the tasks have been spawned. Is there any other way to achieve the same behaviour?